lib/_http_client.js

```diff
@@ -354,7 +354,11 @@ function socketCloseListener() {
       res.emit('aborted');
     }
     req.emit('close');
-    if (res.readable) {
+    if (res.aborted) {
+      res.destroy(null, () => {
+        res.emit('close');
```

Comment:
Instead of duplicating this function here and right below, perhaps we could use a named function in the top scope and reference it in here?

Reply:
fixed